typing/typecore.ml

```diff
@@ -3114,14 +3114,14 @@ let name_cases default lst =
 
 (* Typing of expressions *)
 
-(** [sdesc_for_hint] is used by error messages to report literals in their
+(** [sexp_for_hint] is used by error messages to report literals in their
     original formatting *)
-let unify_exp ?sdesc_for_hint env exp expected_ty =
+let unify_exp ?sexp_for_hint env exp expected_ty =
```

Comment:
The  `sexp_for_hint` is used for more than hints now. I think it would be better to rename it to `sexp` and make it non-optional. That will fix the missing constructors case.

Reply:
Thanks, this fixed many cases indeed.